js/bitopro.js

```diff
@@ -0,0 +1,1151 @@
+        //     {
+        //         "data":[
+        //             {
+        //                 "id":"8384438099",
+        //                 "pair":"bnb_twd",
+        //                 "price":"16000",
+        //                 "avgExecutionPrice":"0",
+        //                 "action":"SELL",
+        //                 "type":"LIMIT",
+        //                 "timestamp":1644902118265,
+        //                 "status":4,
+        //                 "originalAmount":"0.01",
+        //                 "remainingAmount":"0.01",
+        //                 "executedAmount":"0",
+        //                 "fee":"0",
+        //                 "feeSymbol":"twd",
+        //                 "bitoFee":"0",
+        //                 "total":"0",
+        //                 "seq":"BNBTWD8984230567",
+        //                 "updatedTimestamp":1644902118265
+        //             }
+        //         ]
+        //     }
+        //
+        return this.parseTrades (trades, market, since, limit);
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        //
+        //     {
+        //         "serial":"20220215BW14069838",
+        //         "currency":"USDT",
+        //         "protocol":"TRX",
+        //         "address":"TKrwMaZaGiAvtXCFT41xHuusNcs4LPWS7w",
+        //         "amount":"8",
+        //         "fee":"2",
+        //         "total":"10"
+        //     }
+        //
+        const code = this.safeCurrencyCode (this.safeString (transaction, 'coin'));
+        const id = this.safeString (transaction, 'serial');
+        const amount = this.safeNumber (transaction, 'total');
+        const address = this.safeString (transaction, 'address');
+        const fee = this.safeNumber (transaction, 'fee');
+        return {
+            'info': transaction,
+            'id': id,
+            'txid': undefined,
+            'timestamp': undefined,
+            'datetime': undefined,
+            'network': undefined,
+            'addressFrom': undefined,
+            'address': address,
+            'addressTo': address,
+            'tagFrom': undefined,
+            'tag': undefined,
+            'tagTo': undefined,
+            'type': undefined,
+            'amount': amount,
+            'currency': code,
+            'status': undefined,
+            'updated': undefined,
+            'fee': {
+                'currency': code,
+                'cost': fee,
+                'rate': undefined,
+            },
+        };
+    }
+
+    async withdraw (code, amount, address, tag = undefined, params = {}) {
+        [ tag, params ] = this.handleWithdrawTagAndParams (tag, params);
+        await this.loadMarkets ();
+        this.checkAddress (address);
+        const currency = this.currency (code);
+        const request = {
+            'currency': currency['id'],
+            'amount': this.numberToString (amount),
+            'address': address,
+            'protocol': '',
```

Comment:
If this is not a mandatory field, why are we assigning it to an empty string? 
![image](https://user-images.githubusercontent.com/43336371/154246012-20832276-8c3f-4093-85af-f70cc7f378a0.png)

Reply:
I'll comment this line for informational usage.